core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.kt

```diff
@@ -64,15 +65,16 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
 
   private val selectedBooksPath: ArrayList<String>
     get() {
-      val selectedBooksPath = ArrayList<String>()
-      for (item in booksAdapter.items) {
-        if (item.isSelected) {
-          val (_, _, file) = item as BooksOnDiskListItem.BookOnDisk
-          selectedBooksPath.add(file.absolutePath)
-          Log.v(tag, "ZIM PATH : " + file.absolutePath)
+      return booksAdapter.items
+        .filter(BooksOnDiskListItem::isSelected)
+        .filterIsInstance<BookOnDisk>()
+        .map {
+          if (BuildConfig.DEBUG) {
```

Comment:
for things like logging where they aren't actually important to the data I will sometimes use `.also{}`

No change required here, just a personal preference

Reply:
Thanks for the info. I prefer to make this change too. :D

Reply:
https://github.com/kiwix/kiwix-android/pull/1569/commits/4a2d3c4ab92b3ae2be1a9a6585ed9fa109c73fbd